### PR TITLE
Adds a permission check for app directories

### LINKED ISSUE #10154: Integrity check not checking for owner (file system) of apps
### Steps to reproduce
1. Install NC
2. Manually install app, with other user account that "web user"

E.g. my content of apps/ folder looks like this:
```
...
drwxr-xr-x  6 www-data www-data    4096 Jul  5 21:52 provisioning_api/
drwxr-xr-x 10 root     root        4096 Sep 17  2017 rainloop/
drwxr-xr-x 10 www-data www-data    4096 Jul  5 21:52 serverinfo/
...
```

### Expected behaviour
The integrity check should tell me, that `rainloop` is owned by a different owner.

### Actual behaviour
Integrity check shows that everything is OK.

Having a wrong owner results in a weird behavior of the updater: It shows that there is an update. On pressing the update button, the update seems to be performed but when refreshing the page in the browser, again an available update is shown.

### PR DESCRIPTION
**Pull request by weeman1337**
This PR adds a setup check for the permissions of the app root directories.

There is a test for the case the permissions are fine. It's hard to add a test for the wrong permissions, because an ordinary user can't do a `chown`.

Closes #10154

### CODE REVIEW

**Comment by rullzer** [comment]
I think it would be good to get this into 14 as well. As it fixes potential update issues with the update

**Comment by weeman1337** [comment]
Here is a question: I messed around with my git config. For a while the sign off was broken. It didn't complain and added the "signed off" but actually didn't sign it at all o_O

I managed to fix it. Does anyone know a way how to sign off commits in GitHub afterwards?

**Comment by juliushaertl** [comment]
@weeman1337 There is a difference between signing your commits and adding a sign off message. The signoff message is fine for all those commits. The gpg signing seems to be missing, however you would need to do an interactive rebase and edit each commit. You can then just call `git commit --amend --no-edit -S` to sign your commits with gpg.

**Comment by juliushaertl** [comment]
@weeman1337 :grin: Seems you already found it :+1:

**Comment by weeman1337** [comment]
@juliushaertl jep found it - this time it worked. Sometimes it seems that GitHub doesn't recognize updates to already existing commits... ¯\\\_(ツ)\_/¯

File: settings/Controller/CheckSetupController.php [hunk 1]
```diff
@@ -529,6 +531,34 @@ protected function hasOpcacheLoaded(): bool {
...
+	protected function getAppDirsWithDifferentOwner(): array {
+		$currentUser = posix_getpwuid(posix_getuid());
+		$appDirsWithDifferentOwner = [];
+
+		foreach (OC::$APPSROOTS as $appRoot) {
+			$appsPath = $appRoot['path'];
+			$appsDir = new DirectoryIterator($appRoot['path']);
+			foreach ($appsDir as $fileInfo) {
+				if ($fileInfo->isDir() && !$fileInfo->isDot()) {
+					$absAppPath = $appsPath . DIRECTORY_SEPARATOR . $fileInfo->getFilename();
```
Comment (juliushaertl) [inline settings/Controller/CheckSetupController.php]: We should check from the config if the apps directory is marked as writable. If not, there is no reason to report it imho.

**Review by juliusknorr — CHANGES_REQUESTED** [review]

**Comment by weeman1337** [comment]
Ok - I messed it up :( I'll do a new branch/PR including the changes.

**Comment by juliushaertl** [comment]
@weeman1337 seems you tried to pull in recent master changes? A rebase on master and a force push is totally fine inside branches. :wink:

**Comment by juliushaertl** [comment]
Yeah, that is only problematic if multiple people work on the same branch. Otherwise we have very little issues with that. ;)

**Pull request closed without merging**